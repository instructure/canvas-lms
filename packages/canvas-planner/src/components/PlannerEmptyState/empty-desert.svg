<svg width="465" height="220" viewBox="0 0 465 220" xmlns="http://www.w3.org/2000/svg">
  <g transform="translate(0 1)" fill="none" fill-rule="evenodd">
    <path d="M169.49 68.72c1.255-7.153-3.504-13.976-10.65-15.27v-1c-.004-4.95-3.465-9.226-8.306-10.26-4.84-1.034-9.748 1.453-11.774 5.97-4.564-2.72-10.04-3.46-15.162-2.05-5.122 1.41-9.448 4.85-11.978 9.52-3.985-1.048-8.23-.183-11.487 2.34-3.256 2.525-5.153 6.42-5.133 10.54v.21h74.49zM246.14 78.72c-1.012-5.78 2.836-11.29 8.61-12.33v-.78c-.003-4.002 2.793-7.46 6.706-8.298 3.913-.837 7.88 1.175 9.514 4.828 3.683-2.2 8.105-2.805 12.243-1.673 4.14 1.133 7.637 3.903 9.687 7.673 3.22-.867 6.66-.182 9.304 1.852 2.644 2.033 4.188 5.183 4.176 8.518v.17l-60.24.04z" fill="#E6E9EB" fill-rule="nonzero"/>
    <path d="M285 153.84l5-11.77c.098-.23.15-.48.15-.73v-37.51c0-.523.217-1.024.6-1.38l10-9.39h-11.56c-.767 0-1.507.28-2.08.79l-4.32 3.84c-.65.603-1.013 1.454-1 2.34v39.85c.01 1.17-.526 2.28-1.45 3l-4.8 3.74-7 13.94-12.54 7.81-7 1.13c-1.255.204-2.323 1.028-2.84 2.19l-3.34 7.56h56.86l-16.36-11.59c-1.337-.946-1.907-2.65-1.41-4.21l3.09-9.61z" fill="#FEBD96" fill-rule="nonzero"/>
    <path d="M348.51 106.4l1.32-6.2c.127-.6.537-1.1 1.1-1.34l13.57-5.8h-19.16c-1.723 0-3.12 1.397-3.12 3.12v12.42l5.32-.92c.498-.26.854-.73.97-1.28z" fill="#FEC3A0" fill-rule="nonzero"/>
    <path d="M76.71 179.25v-.47l-6.88-11.92c-.12-.212-.28-.4-.47-.55l-3.68-2.86c-.174-.136-.323-.302-.44-.49l-1.89-3c-.19-.3-.29-.646-.29-1v-54.7c0-.802.255-1.584.73-2.23l9.62-13.12H61.07c-.996-.003-1.953.39-2.66 1.09l-3.82 3.82c-.704.706-1.1 1.663-1.1 2.66v67.2c-.004.692-.388 1.326-1 1.65L27 179.25h49.71z" fill="#FEBD96" fill-rule="nonzero"/>
    <path d="M138.39 103.42l-1.77 3.58c-.262.52-.4 1.097-.4 1.68V156h3.09c1.038 0 1.88-.842 1.88-1.88v-42.39c0-.465.17-.914.48-1.26l7.57-8.43c-.633-.453-1.392-.698-2.17-.7h-5.31c-1.427-.002-2.732.803-3.37 2.08z" fill="#FEC3A0" fill-rule="nonzero"/>
    <path d="M242.82 179.25l3.34-7.56c.517-1.162 1.585-1.986 2.84-2.19l7-1.13 12.51-7.84 7-13.94 4.8-3.74c.923-.72 1.46-1.83 1.45-3V100c-.014-.887.35-1.738 1-2.34l4.32-3.84c.573-.51 1.313-.79 2.08-.79h41.37c1.186-.003 2.27.668 2.8 1.73l2.28 4.56c.218.435.33.914.33 1.4v7.85c-.002 1.107.583 2.133 1.538 2.694.954.562 2.135.576 3.102.036.993-.55 1.61-1.595 1.61-2.73V96.19c0-1.723 1.397-3.12 3.12-3.12h24.37c.63 0 1.22.314 1.57.84l3.3 4.94c.2.296.31.643.32 1v54.88c-.003.898.63 1.673 1.51 1.85l4.88 1c.377.064.726.242 1 .51l5.58 5.58c.18.18.394.324.63.42l14.24 5.72 27.8 9.43M0 179.25h464.98M27 179.25l25.52-13.92c.612-.324.996-.958 1-1.65v-67.2c0-.997.396-1.954 1.1-2.66L58.41 90c.706-.704 1.663-1.1 2.66-1.1h44.5c1.323 0 2.55.695 3.23 1.83l5.6 9.33c.353.586.54 1.256.54 1.94v36.83c0 1.038.842 1.88 1.88 1.88H122c1.544.002 2.93.945 3.5 2.38l4.61 11.69c.283.718.978 1.19 1.75 1.19h4.34v-47.3c.002-.583.138-1.158.4-1.68l1.77-3.55c.64-1.275 1.944-2.08 3.37-2.08h5.31c1.663 0 3.13 1.088 3.61 2.68l1.45 4.82c.105.35.16.714.16 1.08v33.51c.002 1.297.67 2.502 1.77 3.19l4.66 2.91 11.65 5.45c.266.124.5.31.68.54l8.19 10.42c.24.305.568.53.94.64l14.29 4.25 4.42 8.43" stroke="#3F5463" stroke-width="2"/>
    <path stroke="#3F5463" stroke-width="2" stroke-linecap="round" d="M316.37 213.87l-2.03-4.93-4.58-11.08 1.17-14.96 7.43-10.57 6.11-5.98M318.36 172.33l5.67-.94M315.09 176.97l-2.1-4"/>
    <path stroke="#3F5463" stroke-width="2" stroke-linecap="round" d="M310.33 190.56l4.76-8.12 4.06-2.24M310.55 187.76l-3.57-5.87M313.88 184.5l4.51.31M313.83 207.72l3.83-14.91 7.92-10.75 8.79-5.6 6.86-2.64 2.94-5.32M310.93 200.69l-4.78-.82M314.34 208.94l10.42 2.9 10.34 4.03 9.79-3.98 8.2-4.07"/>
    <path stroke="#3F5463" stroke-width="2" stroke-linecap="round" d="M336.54 175.63l-.36-3.94 3.84-4.96M327 181.15l-.03-4.74 2.58-4.88 1.88-3.35M316.88 195.84l-3.23-3.3M315.43 201.51l4.38-4.28 6.98-3.84 7.06-6.92M320.44 189.04l5.32-.07 2.86-2.8M332.27 177.8l5.35 1.99 4.14-1.05M314.66 204.5l9.58.27 11.28 1.84 11.81-4.68 5.9-7.09 4.94-5.26"/>
    <path stroke="#3F5463" stroke-width="2" stroke-linecap="round" d="M323.3 195.31l3.56 1.38 6.81.26 4.13-6.44 2.79-6.23 6.46-4.14 2.02-4.16"/>
    <path stroke="#3F5463" stroke-width="2" stroke-linecap="round" d="M329.48 190.75l2.04-4.98 2.3-4.38M322.75 211.28l5.44-2.47 3.19 1.85M343.39 212.5l1.59-3.99 5.71-3.15 2.75-3.91 3.35-2.07M327.3 212.83l1.32 4.45M320.91 204.68l3.16-3.1 7.54 1.03 5.84-1.8 3.85-1.68 4-6 3.5-8.54 3.63-2.35 3.37.62M337.8 190.51l2.81-.35 3.66-3.58"/>
    <path stroke="#3F5463" stroke-width="2" stroke-linecap="round" d="M337.45 200.81l.85-4.74 3.36-3.3M345.88 191.71l3.66-.65 2.86-2.8 2.76.17M335.52 206.61l3.97 2.99M345.35 202.72l1.1-3.37 2.34-4.68M353.23 194.84l4.83.56"/>
    <path d="M238.52 91.16c-5.346 0-9.68 4.334-9.68 9.68v17.35c0 5.42-7 10.75-14.19 13.61v-90c0-5.346-4.334-9.68-9.68-9.68s-9.68 4.334-9.68 9.68V111c-7.22-2.86-14.2-8.19-14.2-13.62V69.6c0-5.346-4.334-9.68-9.68-9.68s-9.68 4.334-9.68 9.68v27.77c0 10.1 5.81 19.49 16.35 26.45 5.27 3.434 11.087 5.946 17.2 7.43v58.67h19.2l.15-37.85c6.113-1.483 11.93-3.996 17.2-7.43 10.55-7 16.35-16.35 16.35-26.44v-17.36c0-5.338-4.322-9.67-9.66-9.68z" fill="#FFF" fill-rule="nonzero"/>
    <path d="M195.29 189.92v-58.67c-6.113-1.484-11.93-3.996-17.2-7.43-10.55-7-16.35-16.35-16.35-26.45V69.6c0-5.346 4.334-9.68 9.68-9.68-4.89 0-4.57 10.06-4.57 10.06v27.75c0 9.75 11.57 23.13 28.45 27.4V41.84c0-2.57 1.023-5.036 2.842-6.852 1.82-1.816 4.287-2.833 6.858-2.828-5.18 0-4.64 8.75-4.64 8.75v149l-5.07.01z" fill="#99E1A2" fill-rule="nonzero"/>
    <path d="M238.52 91.16c-5.346 0-9.68 4.334-9.68 9.68v17.35c0 5.42-7 10.75-14.19 13.61v-90c0-5.346-4.334-9.68-9.68-9.68s-9.68 4.334-9.68 9.68V111c-7.22-2.86-14.2-8.19-14.2-13.62V69.6c0-5.346-4.334-9.68-9.68-9.68s-9.68 4.334-9.68 9.68v27.77c0 10.1 5.81 19.49 16.35 26.45 5.27 3.434 11.087 5.946 17.2 7.43v58.67h19.2l.15-37.85c6.113-1.483 11.93-3.996 17.2-7.43 10.55-7 16.35-16.35 16.35-26.44v-17.36c0-5.338-4.322-9.67-9.66-9.68z" stroke="#3F5463" stroke-width="2"/>
    <path d="M410.38 128.64c3.115 0 5.64 2.525 5.64 5.64v10.11c0 3.16 4.06 6.26 8.27 7.93v-52.4c.127-3.02 2.612-5.403 5.635-5.403 3.023 0 5.508 2.383 5.635 5.403v40.27c4.21-1.67 8.27-4.77 8.27-7.93v-16.17c-.087-2.07.967-4.02 2.747-5.082 1.78-1.06 3.997-1.06 5.776 0 1.78 1.06 2.834 3.012 2.747 5.082v16.17c0 5.88-3.38 11.35-9.52 15.4-3.064 2-6.446 3.463-10 4.33v30.88h-11.3v-18.76c-3.555-.867-6.936-2.33-10-4.33-6.14-4.05-9.52-9.52-9.52-15.4v-10.1c0-3.107 2.513-5.63 5.62-5.64z" fill="#FFF" fill-rule="nonzero"/>
    <path d="M424.28 182.87v-18.76c-3.554-.867-6.936-2.33-10-4.33-6.14-4.05-9.52-8.47-9.52-14.35v-10.52c0-3.115 2.525-5.64 5.64-5.64-2.85 0-2.66 5.86-2.66 5.86v10.51c0 5.68 6.74 12.42 16.57 14.9V99.92c0-3.115 2.525-5.64 5.64-5.64-3 0-2.7 5.1-2.7 5.1v83.16l-2.97.33z" fill="#99E1A2" fill-rule="nonzero"/>
    <path d="M410.38 128.64c3.115 0 5.64 2.525 5.64 5.64v10.11c0 3.16 4.06 6.26 8.27 7.93v-52.4c.127-3.02 2.612-5.403 5.635-5.403 3.023 0 5.508 2.383 5.635 5.403v40.27c4.21-1.67 8.27-4.77 8.27-7.93v-16.17c-.087-2.07.967-4.02 2.747-5.082 1.78-1.06 3.997-1.06 5.776 0 1.78 1.06 2.834 3.012 2.747 5.082v16.17c0 5.88-3.38 11.35-9.52 15.4-3.064 2-6.446 3.463-10 4.33v30.88h-11.3v-18.76c-3.555-.867-6.936-2.33-10-4.33-6.14-4.05-9.52-9.52-9.52-15.4v-10.1c0-3.107 2.513-5.63 5.62-5.64z" stroke="#3F5463" stroke-width="2"/>
    <circle stroke="#CED3D7" stroke-width="2" cx="360.58" cy="18.96" r="11.35"/>
    <path d="M360.58 30.31v7.6M360.58 0v7.6M367.09 28.26l4.36 6.23M349.71 3.43l4.36 6.23M371.25 22.84l7.14 2.6M342.77 12.47l7.14 2.6M371.55 16.02l7.34-1.97M342.27 23.86l7.34-1.96M367.88 10.26l4.89-5.82M348.4 33.48l4.88-5.83" stroke="#CED3D7" stroke-width="2" stroke-linecap="round"/>
    <path d="M431.69 53.54c-.36-2.83-2.085-5.305-4.616-6.624-2.53-1.318-5.548-1.313-8.074.014-2.728-4.45-7.726-6.99-12.93-6.567-5.202.422-9.726 3.735-11.7 8.567-5.62-1.576-11.647.162-15.564 4.488-3.917 4.327-5.05 10.496-2.926 15.932H447c-2.244-7.437-7.95-13.328-15.31-15.81z" fill="#E6E9EB" fill-rule="nonzero"/>
  </g>
</svg>
