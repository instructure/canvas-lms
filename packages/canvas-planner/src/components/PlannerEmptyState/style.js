/*
 * Copyright (C) 2023 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import {getThemeVars} from '../../getThemeVars'

export default function buildStyle() {
  /*
   * If the theme variables to be used when generating the styles below
   * are dependent on the actual theme in use, you can also pull out the
   * `key` property from the return from `getThemeVars()` and do a bit of
   * if or switch statement logic to get the result you want.
   */
  const {variables} = getThemeVars()

  const classNames = {
    root: 'PlannerEmptyState-styles__root',
    small: 'PlannerEmptyState-styles__small',
    desert: 'PlannerEmptyState-styles__desert',
    balloons: 'PlannerEmptyState-styles__balloons',
    subtitlebox: 'PlannerEmptyState-styles__subtitlebox',
    title: 'PlannerEmptyState-styles__title',
    subtitle: 'PlannerEmptyState-styles__subtitle',
  }

  const theme = {
    fontSize: variables.typography.fontSizeMedium,
    fontFamily: variables.typography.fontFamily,
    fontWeight: variables.typography.fontWeightNormal,
    color: variables.colors.oxford,
    background: variables.colors.whiterails,
    lightWeight: variables.typography.fontWeightLight,
    smallSpacing: variables.spacing.small,
    largeSpacing: variables.spacing.large,
    xxLargeSpacing: variables.spacing.xxLarge,
  }

  const css = `
  .${classNames.root} {
    font-size: ${theme.fontSize};
    font-family: ${theme.fontFamily};
    font-weight: ${theme.fontWeight};
    color: ${theme.color};
    background: ${theme.background};
  }
  .${classNames.root}.${classNames.small} .${classNames.desert}, .${classNames.root}.${classNames.small} .${classNames.balloons} {
    max-width: 100%;
  }

  .${classNames.desert}, .${classNames.balloons} {
    display: block;
    margin-inline-start: auto;
    margin-inline-end: auto;
    text-align: center;
    padding: ${theme.smallSpacing};
    margin-top: ${theme.xxLargeSpacing};
    margin-bottom: ${theme.largeSpacing};
  }

  .${classNames.subtitlebox} {
    text-align: center;
    vertical-align: middle;
  }

  .${classNames.title} {
    text-align: center;
    vertical-align: middle;
    padding: 16px;
    font-weight: ${theme.lightWeight};
  }

  .${classNames.subtitle} {
    padding: 12px;
  }
  `

  return {css, classNames, theme}
}
