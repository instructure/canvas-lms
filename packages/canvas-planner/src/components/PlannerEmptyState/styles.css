/* Variables are defined in ./theme.js */

.root {
  font-size: var(--fontSize);
  font-family: var(--fontFamily);
  font-weight: var(--fontWeight);

  color: var(--color);
  background: var(--background);
}

.desert {
  display: block;
  transform: scale(1.4);
  margin-left: auto;
  margin-right: auto;
  text-align: center;
  padding: var(--smallSpacing);
  margin-top: var(--xxLargeSpacing);
  margin-bottom: var(--largeSpacing);
}

.subtitlebox {
  text-align: center;
  vertical-align: middle;
}

.title {
  text-align: center;
  vertical-align: middle;
  padding: 16px;
  font-weight: var(--lightWeight);
}

.subtitle {
  padding: 12px;
}
