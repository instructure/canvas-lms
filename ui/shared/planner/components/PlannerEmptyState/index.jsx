/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React, {Component} from 'react'
import classnames from 'classnames'
import {func, bool, string} from 'prop-types'

import {Heading} from '@instructure/ui-heading'
import {Link} from '@instructure/ui-link'

import {InlineList} from '@instructure/ui-list'
import {useScope as useI18nScope} from '@canvas/i18n'
import DesertSvg from './EmptyDesert' // Currently uses react-svg-loader
import BalloonsSvg from './Balloons'
import buildStyle from './style'

const I18n = useI18nScope('planner')

export default class PlannerEmptyState extends Component {
  static propTypes = {
    changeDashboardView: func,
    onAddToDo: func.isRequired,
    isCompletelyEmpty: bool,
    responsiveSize: string,
    isWeekly: bool,
  }

  static defaultProps = {
    responsiveSize: 'large',
  }

  constructor(props) {
    super(props)
    this.style = buildStyle()
  }

  handleDashboardCardLinkClick = () => {
    if (this.props.changeDashboardView) {
      this.props.changeDashboardView('cards')
    }
  }

  renderAddToDoButton = () => {
    return (
      <Link
        as="button"
        isWithinText={false}
        id="PlannerEmptyState_AddToDo"
        onClick={this.props.onAddToDo}
      >
        {I18n.t('Add To-Do')}
      </Link>
    )
  }

  renderNothingAtAll = () => {
    return (
      <div
        className={classnames(
          this.style.classNames.root,
          'planner-empty-state',
          this.style.classNames[this.props.responsiveSize]
        )}
      >
        <DesertSvg
          className={classnames(this.style.classNames.desert, 'desert')}
          aria-hidden="true"
        />
        <div className={this.style.classNames.title}>
          <Heading>{I18n.t('No Due Dates Assigned')}</Heading>
        </div>
        <div className={this.style.classNames.subtitlebox}>
          <div className={this.style.classNames.subtitle}>
            {I18n.t("Looks like there isn't anything here")}
          </div>
          {!this.props.isWeekly && (
            <InlineList delimiter="pipe">
              {this.props.changeDashboardView && (
                <InlineList.Item>
                  <Link
                    as="button"
                    isWithinText={false}
                    id="PlannerEmptyState_CardView"
                    onClick={this.handleDashboardCardLinkClick}
                  >
                    {I18n.t('Go to Card View Dashboard')}
                  </Link>
                </InlineList.Item>
              )}
              <InlineList.Item>{this.renderAddToDoButton()}</InlineList.Item>
            </InlineList>
          )}
        </div>
      </div>
    )
  }

  renderNothingLeft = () => {
    const msg = this.props.isWeekly ? I18n.t('Nothing Due This Week') : I18n.t('Nothing More To Do')
    return (
      <div
        className={classnames(
          this.style.classNames.root,
          'planner-empty-state',
          this.style.classNames[this.props.responsiveSize]
        )}
      >
        <BalloonsSvg
          className={classnames(this.style.classNames.balloons, 'balloons')}
          aria-hidden="true"
        />
        <div className={this.style.classNames.title}>
          <Heading>{msg}</Heading>
        </div>
        {!this.props.isWeekly && (
          <div className={this.style.classNames.subtitlebox}>
            <div className={this.style.classNames.subtitle}>
              {I18n.t('Scroll up to see your history!')}
            </div>
            {this.renderAddToDoButton()}
          </div>
        )}
      </div>
    )
  }

  render() {
    return (
      <>
        <style>{this.style.css}</style>
        {this.props.isCompletelyEmpty ? this.renderNothingAtAll() : this.renderNothingLeft()}
      </>
    )
  }
}
