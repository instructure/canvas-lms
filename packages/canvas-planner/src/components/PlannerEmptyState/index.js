/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that they will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React, { Component } from 'react';
import themeable from '@instructure/ui-themeable/lib';
import { func } from 'prop-types';

import Heading from '@instructure/ui-core/lib/components/Heading';
import Link from '@instructure/ui-core/lib/components/Link';

import formatMessage from '../../format-message';
import DesertSvg from './empty-desert.svg'; // Currently uses react-svg-loader

import styles from './styles.css';
import theme from './theme.js';

class PlannerEmptyState extends Component {

  static propTypes = {
    changeToDashboardCardView: func
  }

  handleDashboardCardLinkClick = () => {
    if (this.props.changeToDashboardCardView) {
        this.props.changeToDashboardCardView();
    }
  }

  render () {
    return (
      <div className={styles.root}>
        <DesertSvg className={styles.desert} aria-hidden="true" />
        <div className={styles.title}>
          <Heading>{formatMessage("No Due Dates Assigned")}</Heading>
        </div>
        <div className={styles.subtitlebox}>
          <div className={styles.subtitle}>{formatMessage("Looks like there isn't anything here")}</div>
          <Link onClick={this.handleDashboardCardLinkClick}>{formatMessage("Go to Dashboard Card View")}</Link>
        </div>
      </div>
    );
  }
}

export default themeable(theme, styles)(PlannerEmptyState);
