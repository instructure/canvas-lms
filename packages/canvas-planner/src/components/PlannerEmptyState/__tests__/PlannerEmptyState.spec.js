/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that they will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react';
import { shallow, mount } from 'enzyme';
import PlannerEmptyState from '../index';

it('renders empty page', () => {
  const wrapper = shallow(<PlannerEmptyState changeToDashboardCardView={() => {}} />, );
  expect(wrapper).toMatchSnapshot();
});

it('does not changeToDashboardCardView on mount', () => {
  const mockDispatch = jest.fn();

  const changeToDashboardCardView = mockDispatch;

  mount(<PlannerEmptyState changeToDashboardCardView={changeToDashboardCardView} />, );
  expect(changeToDashboardCardView).not.toHaveBeenCalled();
});

it('calls changeToDashboardCardView on link click', () => {
  const mockDispatch = jest.fn();

  const changeToDashboardCardView = mockDispatch;

  const wrapper = mount(<PlannerEmptyState changeToDashboardCardView={changeToDashboardCardView} />, );
  const button = wrapper.find('Link').find('button');

  button.simulate('click');
  expect(changeToDashboardCardView).toHaveBeenCalled();
});

it('does not call changeToDashboardCardView on false prop', () => {
  const wrapper = mount(<PlannerEmptyState/>, );
  const button = wrapper.find('Link').find('button');

  button.simulate('click');
  expect(() => {
    button.simulate('click');
  }).not.toThrow();
});
