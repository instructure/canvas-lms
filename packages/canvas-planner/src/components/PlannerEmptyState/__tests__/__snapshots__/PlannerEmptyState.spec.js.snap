// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty page 1`] = `
<div>
  <DesertSvg
    aria-hidden="true"
  />
  <div>
    <Heading
      border="none"
      color="inherit"
      ellipsis={false}
      level="h2"
    >
      No Due Dates Assigned
    </Heading>
  </div>
  <div>
    <div>
      Looks like there isn't anything here
    </div>
    <Link
      as="button"
      ellipsis={false}
      linkRef={[Function]}
      onClick={[Function]}
      variant="default"
    >
      Go to Dashboard Card View
    </Link>
  </div>
</div>
`;
